`build_charms.yaml` fails requesting password

### Bug Description

The full context is in https://github.com/canonical/data-platform-workflows/pull/259, TL;DR:

When building charms, the `Set up environment` step from `build_charms.yaml`, it fails with:

```
info: Adding user `runner' to group `lxd' ...
Password: Invalid password.
Error: Process completed with exit code 1.
```

### To Reproduce

1. Use `canonical/charmed-kubeflow-workflows/.github/workflows/get_charms_build_with_cache.yaml@main` in any CI workflow
2. Run the CI

### Environment

CI environment of the https://github.com/canonical/kfp-operators/pull/640 PR and development branch

### Relevant Log Output

```shell
info: Adding user `runner' to group `lxd' ...
Password: Invalid password.
Error: Process completed with exit code 1.
```

### Additional Context

Failing for https://github.com/canonical/kfp-operators/actions/runs/12637402786/job/35211538308?pr=640#step:9:31

--- a/.github/workflows/get_charms_build_with_cache.yaml
+++ b/.github/workflows/get_charms_build_with_cache.yaml
@@ -20,7 +20,7 @@ jobs:
 
   build:
     name: Build charms
-    uses: canonical/data-platform-workflows/.github/workflows/build_charm.yaml@v21.0.0
+    uses: canonical/data-platform-workflows/.github/workflows/build_charm.yaml@v24.0.5
     needs: get-charm-paths
     strategy:
       fail-fast: false

